### PR TITLE
Module history endpoint + Fixes to ModuleState query

### PR DESCRIPTION
**Pull request by jonathanwgoodwin**
@gchomatas 

This pr fixes an issue with the ModuleState query where it was returning no results for a module if it did not have both a "last successful build" and a "last non skipped build".

This pr makes a change to the module history endpoint to paginate it. The endpoint returns a `ModuleActivityPage` which holds the results and the number of remaining after that page.

### CODE REVIEW

File: BlazarData/src/main/java/com/hubspot/blazar/data/service/ModuleBuildService.java [hunk 1]
```diff
@@ -47,8 +51,32 @@ public ModuleBuildService(ModuleBuildDao moduleBuildDao, ModuleDao moduleDao, Ev
     return moduleBuildDao.getByState(state);
   }
 
-  public List<ModuleBuild> getByModule(int moduleId) {
-    return moduleBuildDao.getByModule(moduleId);
+  public ModuleActivityPage getModuleBuildHistoryPage(int moduleId, Optional<Integer> maybeFromBuildNumber, Optional<Integer> maybePageSize) {
```
Comment (andyhuang91) [inline BlazarData/src/main/java/com/hubspot/blazar/data/service/ModuleBuildService.java]: Method name still has 'History' in it instead of activity. Maybe the dao methods should be changed too

File: BlazarData/src/main/java/com/hubspot/blazar/data/service/ModuleBuildService.java [hunk 2]
```diff
@@ -47,8 +51,32 @@ public ModuleBuildService(ModuleBuildDao moduleBuildDao, ModuleDao moduleDao, Ev
...
+      if (maxBuildNumber.isPresent()) {
+        fromBuildNumber = maxBuildNumber.get();
+      } else {
+        return new ModuleActivityPage(ImmutableList.of(), 0);
+      }
+    }
+
+    // get a specific page
+    long start = System.currentTimeMillis();
+    List<ModuleBuildInfo> builds = moduleBuildDao.getLimitedModuleBuildHistory(moduleId, fromBuildNumber, pageSize);
```
Comment (andyhuang91) [inline BlazarData/src/main/java/com/hubspot/blazar/data/service/ModuleBuildService.java]: This means we have 3 separate db requests if `fromBuildNumber` is not specified?

File: BlazarService/src/main/java/com/hubspot/blazar/resources/RepositoryBuildResource.java [hunk 3]
```diff
@@ -22,20 +22,24 @@
...
 import com.hubspot.blazar.data.service.ModuleBuildService;
 import com.hubspot.blazar.data.service.RepositoryBuildService;
+import com.hubspot.blazar.data.service.StateService;
 
 @Path("/branches/builds")
 @Produces(MediaType.APPLICATION_JSON)
 public class RepositoryBuildResource {
   private final RepositoryBuildService repositoryBuildService;
   private final ModuleBuildService moduleBuildService;
+  private StateService stateService;
```
Comment (andyhuang91) [inline BlazarService/src/main/java/com/hubspot/blazar/resources/RepositoryBuildResource.java]: Unused?

File: BlazarBase/src/main/java/com/hubspot/blazar/base/ModuleState.java [hunk 4]
```diff
@@ -11,46 +11,46 @@
 
   private final Module module;
   private final Optional<ModuleBuild> lastSuccessfulModuleBuild;
-  private final Optional<RepositoryBuild> lastSuccessfulRepoBuild;
+  private final Optional<RepositoryBuild> lastSuccessfulBranchBuild;
```
Comment (andyhuang91) [inline BlazarBase/src/main/java/com/hubspot/blazar/base/ModuleState.java]: Would it make sense now to have ModuleState contain ModuleBuildInfos instead of separate fields for module and branch (repo) builds?

File: BlazarData/src/main/java/com/hubspot/blazar/data/dao/StateDao.java [hunk 5]
```diff
@@ -40,39 +41,52 @@
...
+   */
   @SqlQuery("" +
       "SELECT * " +
-      "  FROM modules AS module " +
+      "  FROM branches " +
+      "     LEFT OUTER JOIN modules as module on (branches.id = module.branchId) " +
       // last build
       "     LEFT OUTER JOIN module_builds AS lastModuleBuild ON (module.lastBuildId = lastModuleBuild.id) " +
-      "     LEFT OUTER JOIN repo_builds AS lastRepoBuild ON (lastModuleBuild.repoBuildId = lastRepoBuild.id) " +
+      "     LEFT OUTER JOIN repo_builds AS lastBranchBuild ON (lastModuleBuild.repoBuildId = lastRepoBuild.id) " +
```
Comment (andyhuang91) [inline BlazarData/src/main/java/com/hubspot/blazar/data/dao/StateDao.java]: This query is wrong `repo_builds AS lastBranchBuild` but you have `lastRepoBuild.id`

File: BlazarBase/src/main/java/com/hubspot/blazar/base/ModuleState.java (hunk 4 above)
Comment (jonathanwgoodwin) [inline BlazarBase/src/main/java/com/hubspot/blazar/base/ModuleState.java]: I agree it would make things a bit more standardized, and would probably make sense to return something like that to the FE. I think though I still need this as an intermediate representation, because of how rosetta creates objects from the mysql results.

Would you prefer something like:
```
{'lastSuccessful': { 
  'moduleBuild': <>, 
  'branchBuild': <>
}}
```

File: BlazarData/src/main/java/com/hubspot/blazar/data/service/ModuleBuildService.java (hunk 1 above)
Comment (jonathanwgoodwin) [inline BlazarData/src/main/java/com/hubspot/blazar/data/service/ModuleBuildService.java]: Yeah I'll refactor this

File: BlazarBase/src/main/java/com/hubspot/blazar/base/ModuleState.java (hunk 4 above)
Comment (jonathanwgoodwin) [inline BlazarBase/src/main/java/com/hubspot/blazar/base/ModuleState.java]: We agreed to do this in a separate PR so we can start User testing this page.
